zcash_primitives/src/transaction/builder.rs

```diff
@@ -622,58 +250,47 @@ impl<'a, P: consensus::Parameters, R: RngCore> Builder<'a, P, R> {
         self.progress_notifier = Some(progress_notifier);
     }
 
+    pub fn value_balance(&self) -> Result<Amount, Error> {
```

Comment:
I don't think this is a useful API to expose. It doesn't account for the fee (which users can't alter, by design), so it can't be directly used during note selection (and I don't see why a wallet would incrementally add notes to the builder to keep track of balance, instead of just selecting the note values themselves from however they store them).
```suggestion
    fn value_balance(&self) -> Result<Amount, Error> {
```